application-ckeditor-plugins/src/main/resources/xwiki-upload/plugin.js

```diff
@@ -139,4 +141,28 @@
     //   see https://ckeditor.com/docs/ckeditor4/latest/guide/dev_file_upload.html#request-2
     }, null, null, 1);
   };
+
+  var listenUploadRequest = function (editor) {
+    editor.on('fileUploadRequest', function (event) {
+      if (editor.config['xwiki-upload']) {
+        var xhr = event.data.fileLoader.xhr;
+        xhr.setRequestHeader( 'X-XWiki-Temporary-Attachment-Support',
+          editor.config['xwiki-upload'].isTemporaryAttachmentSupported);
+      }
+    });
+  };
+
+  // Inject a new input field when an attachment is added so that the save request knows which are the new attachments.
+  var listenToUploadedAttachments = function(editor) {
+    editor.on('fileUploadResponse', function (event) {
+      if ((editor.config['xwiki-upload'] || {}).isTemporaryAttachmentSupported) {
```

Comment:
You don't do this fail-safe above. Either it's not needed and thus you can remove it or it's needed and you need to do it above also. BTW, I think you're missing https://github.com/xwiki-contrib/application-ckeditor/blob/master/application-ckeditor-plugins/src/main/resources/xwiki-macro/plugin.js#L24 for the 'xwiki-upload' config.

Reply:
>  BTW, I think you're missing https://github.com/xwiki-contrib/application-ckeditor/blob/master/application-ckeditor-plugins/src/main/resources/xwiki-macro/plugin.js#L24 for the 'xwiki-upload' config.

Right. 

> it's needed and you need to do it above also

Not sure it's needed anymore, since the upgrade of CKEditor will provide it. So I guess it's safe without it now.